lib/cisco_node_utils/ace.rb

```diff
@@ -0,0 +1,225 @@
+                                option_format: '' }
+      @regexp_str = '(?<seqno>\d+) (?<action>\S+)'\
+                 ' *(?<proto>\d+|\S+)'\
+                 ' *(?<src_addr>any|host \S+|\S+\/\d+|\S+ [:\.0-9a-fA-F]+|'\
+                 'addrgroup \S+)*'\
+                 ' *(?<src_port>eq \S+|neq \S+|lt \S+|''gt \S+|range \S+ \S+|'\
+                 'portgroup \S+)?'\
+                 ' *(?<dst_addr>any|host \S+|\S+\/\d+|\S+ [:\.0-9a-fA-F]+|'\
+                 'addrgroup \S+)'\
+                 ' *(?<dst_port>eq \S+|neq \S+|lt \S+|gt \S+|range \S+ \S+|'\
+                 'portgroup \S+)?'\
+                 ' *(?<option>[a-zA-Z0-9\-\/ ]*)*'
+    end
+
+    # Create a hash of all aces under give acl_name.
+    def self.aces
+      regexp_str = '(?<seqno>\d+) (?<action>\S+)'\
+                 ' *(?<proto>\d+|\S+)'\
+                 ' *(?<src_addr>any|host \S+|\S+\/\d+|\S+ [:\.0-9a-fA-F]+|'\
+                 'addrgroup \S+)*'\
+                 ' *(?<src_port>eq \S+|neq \S+|lt \S+|''gt \S+|range \S+ \S+|'\
+                 'portgroup \S+)?'\
+                 ' *(?<dst_addr>any|host \S+|\S+\/\d+|\S+ [:\.0-9a-fA-F]+|'\
+                 'addrgroup \S+)'\
+                 ' *(?<dst_port>eq \S+|neq \S+|lt \S+|gt \S+|range \S+ \S+|'\
+                 'portgroup \S+)?'\
+                 ' *(?<option>[a-zA-Z0-9\-\/ ]*)*'
+      afis = %w(ip ipv6)
+      hash = {}
+      afis.each do |afi|
+        get_args = { afi: afi }
+        instances = config_get('acl', 'all_acl', get_args)
+        next if instances.nil?
+        instances.each do |name|
+          hash[name] = {}
+          get_args[:acl_name] = name
+          aces = config_get('acl', 'all_ace', get_args)
+          next if aces.nil?
+          aces.each do |ace|
+            regexp = Regexp.new(regexp_str)
+            match = regexp.match(ace)
+            item = Ace.new(name, match[1].to_i, afi)
+            item.seqno = match[1].to_i
+            item.action = match[2]
+            item.proto = match[3]
+            item.src_addr = match[4]
+            item.src_port = match[5]
+            item.dst_addr = match[6]
+            item.dst_port = match[7]
+            item.option_format = match[8]
+            hash[name][match[1].to_i] = item
+          end
+        end
+      end
+      hash
+    end
+
+    # Destroy a router instance; disable feature on last instance
+    def destroy
+      config_ace('no')
+    end
+
+    # First create acl
+    def config_ace(state='')
+      @set_args[:state] = state
+      config_set('acl', 'ace', @set_args)
+    end
+
+    # ----------
+    # PROPERTIES
+    # ----------
+    # getter of seqno
+    def seqno
+      str = config_get('acl', 'ace', @get_args)
+      return if str.nil?
+
+      regexp = Regexp.new(@regexp_str)
+      match = regexp.match(str)
+      match[1]
+    end
+
+    # setter of seqno
+    def seqno=(state)
+      @set_args[:state] = state ? '' : 'no'
+    end
+
+    # getter of action
+    def action
```

Comment:
Replace `action`, `proto`, `src_addr`, `src_port`, `dst_addr`, `dst_port` getters & setters with the method examples I showed in the earlier comment.